Engg 2201 code refactor for using the configurations and prompts from db

saran-damm opened the pull request.

saran-damm committed:
Inital changes for SQLAlchemy integration

saran-damm committed:
ENGG-2199 Changes for SQLAlchemy Integration

saran-damm committed:
ENGG-2199 Agent and Base changes

saran-damm committed:
ENGG-2200: Add prompt templates and groups models

The changes in this commit add new database models for prompt templates and prompt template groups. The `PromptTemplates` model represents individual prompt templates, including their name, template content, version, creation and update timestamps, and a flag to indicate the current version. The `PromptTemplatesGroups` model represents groups of prompt templates, with a name and a dictionary of variables. The `PromptTemplatesGroupsMapping` model provides a many-to-many relationship between prompt templates and groups.

These changes are part of a larger effort to introduce a more flexible and extensible prompt management system, allowing for the creation, versioning, and grouping of prompt templates.

saran-damm committed:
ENGG-2200: Add prompt template read functions

This commit adds two new functions to the `prompts.py` module:

1. `read_prompt_template_by_name`: This function reads a prompt template by its name from the database and returns the selected columns as a dictionary.

2. `read_prompt_template_by_prompts_id`: This function reads all prompt templates associated with a given prompts ID by joining the `PromptTemplates` and `PromptTemplatesGroupsMapping` tables. It returns the selected columns as a list of dictionaries.

These functions are added to provide a convenient way to fetch prompt template data from the database, which is necessary for the application's functionality.

gitbutler-client committed:
GitButler Workspace Commit

This is a merge commit the virtual branches in your workspace.

Due to GitButler managing multiple virtual branches, you cannot switch back and
forth between git branches and virtual branches easily. 

If you switch to another branch, GitButler will need to be reinitialized.
If you commit on this branch, GitButler will throw it away.

Here are the branches that are currently applied:
 - ENGG-2200-db-orm-using-pydantic (refs/gitbutler/ENGG-2200-db-orm-using-pydantic)
   branch head: 7ff8dc937dc83eaa7979a790c26c1bcd3ae2d38b
   - db/psql_tables.py
   - backend/server.py
   - db/db_utils.py
   - backend/entities.py
   - db/prompts.py
   - db/get_prompts.py
For more information about what we're doing here, check out our docs:
https://docs.gitbutler.com/features/virtual-branches/integration-branch

saran-damm committed:
feat: Add prompts_id field to AgentConfig and ChatAgents models

The changes made in this commit include:

1. Adding a new field `prompts_id` to the `AgentConfig` and `ChatAgents` models. This field will store the ID of the prompt templates associated with the agent configuration.

2. Updating the `get_agent` function in `db/db_utils.py` to include the `prompts_id` field in the list of columns being read from the database.

3. Creating a new module `db/get_prompts.py` that contains functions to read prompt templates by name and by ID. This will be used to retrieve the prompt templates associated with the agent configuration.

These changes are necessary to support the ability to associate different prompt templates with different agent configurations, allowing for more flexibility and customization in the application.

saran-damm committed:
feat: Add prompts_from_db parameter to GPTResearcher and related classes

    
The changes add a new parameter `prompts_from_db` to the `GPTResearcher` class and related functions in the `multi_agents/agents/researcher.py`, `gpt_researcher/master/actions.py`, and `backend/websocket_manager.py` files. This parameter allows the caller to pass in prompts from a database, which can be used in the research process.
    
The main motivation for this change is to provide more flexibility and customization options for the research process, allowing users to incorporate their own pre-defined prompts or templates into the research workflow.

saran-damm committed:
feat: Add prompts_from_db parameter to subtopic research

The changes add a new parameter `prompts_from_db` to the `run_subtopic_research` method of the `ResearchAgent` class. This parameter is used to pass a list of prompts from the database to the research process, which can be used to generate more comprehensive and relevant reports.

rohitashtanwar commented on db/db_utils.py:
```diff
@@ -0,0 +1,31 @@
+
+logger = logging.getLogger(__name__)
+
+async def get_agent(psql_sess: AsyncSession, base_id: Union[str, None], agent_id: str = None,
```
Since we are not using `base_id`, can we remove this?

rohitashtanwar commented on db/db_utils.py:
```diff
@@ -0,0 +1,31 @@
+logger = logging.getLogger(__name__)
+
+async def get_agent(psql_sess: AsyncSession, base_id: Union[str, None], agent_id: str = None,
+                    agent_name: Union[str, None] = "splore") -> Tuple[dict, str, str, Union[str, None]]:
```
Since we are not using `base_id` and `agent_name`, can we remove these?

saran-damm committed:
ENGG-2201: Remove task from research report

Removes the 'task' field from the research report returned by the ChiefEditorAgent. This change is made to avoid unnecessary data being transmitted to the client.

Commit:
updated the deployment
